kafka-rest/src/test/java/io/confluent/kafkarest/controllers/RecordSerializerFacadeTest.java

```diff
@@ -802,30 +803,90 @@ public void serializeNullAvroKeyNullSchema_returnsEmpty() {
+        "{\"type\":\"record\",\"name\":\"MyRecord\",\"namespace\":\"com.example\",\""
+            + "fields\":[{\"name\":\"myNullableInt\","
+            + "\"type\":[\"null\",\"int\"],\"default\":null}]}";
+    AvroSchema schema = new AvroSchema(schemaString);
+    String subject = SUBJECT_NAME_STRATEGY.subjectName(TOPIC_NAME, /* isKey= */ true, schema);
+    int schemaId = schemaRegistryClient.register(subject, schema);
+    ObjectNode node = new ObjectNode(JsonNodeFactory.instance);
+    node.putNull("myNullableInt");
+    ByteString serialized =
+        recordSerializer
+            .serialize(
+                EmbeddedFormat.AVRO,
+                TOPIC_NAME,
+                Optional.of(RegisteredSchema.create(subject, schemaId, SCHEMA_VERSION, schema)),
+                node,
+                /* isKey= */ true)
+            .get();
+
+    KafkaAvroDeserializer deserializer = new KafkaAvroDeserializer();
+    deserializer.configure(SCHEMA_SERIALIZER_CONFIGS, /* isKey= */ true);
+    GenericRecord expectedRecord = new GenericData.Record(schema.rawSchema());
+    expectedRecord.put("myNullableInt", null);
+    Object deserialized =
+        deserializer.deserialize(TOPIC_NAME, serialized.toByteArray(), schema.rawSchema());
+    assertEquals(expectedRecord, deserialized);
+  }
+
+  @Test
+  public void serializeNullAvroKeyNonNullableSchema_throwsBadRequestException() throws Exception {
     AvroSchema schema = new AvroSchema("{\"type\": \"int\"}");
     String subject = SUBJECT_NAME_STRATEGY.subjectName(TOPIC_NAME, /* isKey= */ true, schema);
     int schemaId = schemaRegistryClient.register(subject, schema);
 
-    Optional<ByteString> serialized =
+    BadRequestException badRequestException =
+        assertThrows(
+            BadRequestException.class,
+            () ->
+                recordSerializer.serialize(
+                    EmbeddedFormat.AVRO,
+                    TOPIC_NAME,
+                    Optional.of(RegisteredSchema.create(subject, schemaId, SCHEMA_VERSION, schema)),
+                    NullNode.getInstance(),
+                    /* isKey= */ true));
+
+    assertEquals("Expected int. Got VALUE_NULL", badRequestException.getMessage());
+  }
+
+  @Test
+  public void serializeNullAvroKeyNonNullableSchemaAlwaysEmptyConfigured_returnsEmpty()
+      throws Exception {
+    AvroSchema schema = new AvroSchema("{\"type\": \"int\"}");
+    String subject = SUBJECT_NAME_STRATEGY.subjectName(TOPIC_NAME, /* isKey= */ true, schema);
+    int schemaId = schemaRegistryClient.register(subject, schema);
+
+    recordSerializer =
+        new RecordSerializerFacade(
+            new NoSchemaRecordSerializer(SCHEMA_SERIALIZER_CONFIGS),
+            () ->
+                new SchemaRecordSerializerImpl(
+                    schemaRegistryClient,
+                    SCHEMA_SERIALIZER_CONFIGS,
+                    SCHEMA_SERIALIZER_CONFIGS,
+                    SCHEMA_SERIALIZER_CONFIGS,
+                    true));
```

Comment:
cool, thanks for adding the case when we flip the config to true and use the old behavior